src/VisualStudio/Core/Def/Implementation/DesignerAttribute/DesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -1,339 +0,0 @@
-            if (ErrorHandler.Failed(vsProject.GetItemContext((uint)VSConstants.VSITEMID.Root, out var projectServiceProvider)))
-            {
-                _cpsProjects.TryAdd(project.Id, null);
-                return null;
-            }
-
-            var serviceProvider = new Shell.ServiceProvider(projectServiceProvider);
-            var updateService = serviceProvider.GetService(typeof(IProjectItemDesignerTypeUpdateService)) as IProjectItemDesignerTypeUpdateService;
-            if (updateService == null)
-            {
-                _cpsProjects.TryAdd(project.Id, null);
-                return null;
-            }
-
-            _cpsProjects[project.Id] = updateService;
-            return updateService;
-        }
-
-        private bool CheckVersions(Document document, VersionStamp textVersion, VersionStamp semanticVersion, Data existingData)
-        {
-            // first check full version to see whether we can reuse data in same session, if we can't, check timestamp only version to see whether
-            // we can use it cross-session.
-            return document.CanReusePersistedTextVersion(textVersion, existingData.TextVersion) &&
-                   document.Project.CanReusePersistedDependentProjectVersion(semanticVersion, existingData.SemanticVersion);
-        }
-
-        private async Task RegisterDesignerAttributeAndSaveStateAsync(
-            Document document, VersionStamp textVersion, VersionStamp semanticVersion, Optional<string> designerAttributeArgumentOpt, CancellationToken cancellationToken)
-        {
-            if (!designerAttributeArgumentOpt.HasValue)
-            {
-                // no value means it couldn't determine whether this document has designer attribute or not.
-                // one of such case is when base type is error type.
-                return;
-            }
-
-            var data = new Data(textVersion, semanticVersion, designerAttributeArgumentOpt.Value);
-            await _state.PersistAsync(document, data, cancellationToken).ConfigureAwait(false);
-
-            await RegisterDesignerAttributeAsync(document, designerAttributeArgumentOpt.Value, cancellationToken).ConfigureAwait(false);
-        }
-
-        private async Task RegisterDesignerAttributeAsync(Document document, string designerAttributeArgument, CancellationToken cancellationToken)
-        {
-            if (!_state.Update(document.Id, designerAttributeArgument))
-            {
-                // value is not updated, meaning we are trying to report same value as before.
-                // we don't need to do anything. 
-                //
-                // I kept this since existing code had this, but since we check what platform has before
-                // reporting it, this might be redundant.
-                return;
-            }
-
-            if (!(document.Project.Solution.Workspace is VisualStudioWorkspaceImpl workspace))
-            {
-                return;
-            }
```

Comment:
the logic below here was preserved (i.e. how we notify CPS and legacy projects of this attribute value).